### DISCUSSION TITLE
Migration Script

### DISCUSSION THREAD

**Discussion by rossanodr** [chakra-ui/chakra-ui#8565] [Ideas]
First of all: chakra is amazing. Thanks for the amazing job you've done.

As we continually strive to enhance our projects with the latest updates and features from Chakra UI, it would be incredibly helpful to have an automated migration assistance script. This tool could simplify the process of updating imports and adapting to breaking changes when upgrading between major versions of Chakra UI.

For instance, the team at LangChain has a similar utility in the form of 
@langchain/scripts

. This script effectively handles the upgrade of imports and other related tasks for their library. You can find more details about it here: https://js.langchain.com/v0.2/docs/versions/v0_2/.

Implementing something similar for Chakra UI would save developers a significant amount of time and reduce the risk of manual errors during the upgrade process. It would be a valuable addition to your already fantastic toolkit.

Thank you for considering this feature!

**Comment by AkshayHere** [comment]
Guess no one want to talk about this now.

**Reply by rossanodr** [reply]
> Guess no one want to talk about this now.

for me it was easier to migrate to shadcn :)

**Comment by jamesopti** [comment]
I agree it would be nice to have a much more clear migration path from Chakra v2 to v3.

We have a quite large React app for our business in production that there's no way we can migrate everything in one PR.

I've been searching for ways to do this. #9853 is very related.

**Comment by segunadebayo** [comment]
The upgrade codemod is in `@chakra-ui/cli`. Run `npx @chakra-ui/cli upgrade`.
